src/backend/util/uniforms_buffer.rs

```diff
@@ -1,5 +1,13 @@
 use super::buffer::create_buffer_with_value;
 
+pub unsafe trait ToStd140 {
```

Comment:
Because the trait implies that `Output` is layout compatible with the relevant struct from the shader. If it is not, it will cause the shader to access the wrong memory.

Although I'm not 100% sure if that can cause memory safety issues for the Rust program, if I'm honest.

Reply:
Okay interesting!